### PR TITLE
fix(media): Add 'ended' event handler on MediaStreamTrack for FF

### PR DESCRIPTION
**Pull request by jallamsetty1**
When screensharing is stopped on FF from the sharing indicator (available in FF nightly), we are currently not detecting it.
Firefox fires a track 'ended' event as opposed to Chrome which fires a mediastream 'inactive' event as well.

### CODE REVIEW

File: modules/RTC/JitsiTrack.js [hunk 1]
```diff
@@ -134,7 +140,7 @@ export default class JitsiTrack extends EventEmitter {
             }
         }
         if (this._streamInactiveHandler) {
-            addMediaStreamInactiveHandler(this.stream, undefined);
+            this._addMediaStreamInactiveHandler(undefined);
```
Comment (fippo) [inline modules/RTC/JitsiTrack.js:143]: this is a bit confusing, why isn't the streamInactiveHandler passed here as well?

Comment (jallamsetty1) [reply modules/RTC/JitsiTrack.js:143]: we are unregistering the handler here when the stream is getting removed from the conference.

**Comment by jallamsetty1** [comment]
> i'd use track.onended in all browsers. stream.oninactive can happen for other (arguably unlikely) reasons such as all tracks getting removed from the stream.

Thanks for the review! I agree we should be using track.onended but this is something I haven't tested in the rest of the browsers yet. There is more refactoring planned around local/remote tracks and I will make this change then.

File: modules/RTC/JitsiTrack.js (hunk 1 above)
Comment (fippo) [reply modules/RTC/JitsiTrack.js:143]: ah. and it uses .onended, not addEventListener('ended') so that works. :+1:

**Pull request merged**